src/Libraries/Microsoft.Extensions.Http.Telemetry/Logging/IHttpClientLogEnricher.cs

```diff
@@ -17,5 +18,8 @@ public interface IHttpClientLogEnricher
     /// <param name="collector">Tag collector to add tags to.</param>
     /// <param name="request"><see cref="HttpRequestMessage"/> object associated with the outgoing HTTP request.</param>
     /// <param name="response"><see cref="HttpResponseMessage"/> object associated with the outgoing HTTP request.</param>
-    void Enrich(IEnrichmentTagCollector collector, HttpRequestMessage? request = null, HttpResponseMessage? response = null);
```

Comment:
@geeknoid @RussKie @dariusclay @dpk83 what do you think folks?
- is it better to keep the existing method `Enrich(IEnrichmentTagCollector, HttpRequestMessage, HttpResponseMessage)` and add `Enrich(IEnrichmentTagCollector collector, Exception, HttpRequestMessage, HttpResponseMessage)`; or
- extend the existing method to have `void Enrich(IEnrichmentTagCollector, HttpRequestMessage, HttpResponseMessage?, Exception?)` only?
What will be better for users (and better for backward-compat with 1P services)?

Reply:
I think I'd rather stick to one method and add the optional exception.

Reply:
The main question is whether the changes are breaking or not. Once we GA, we can't have API breaking changes. Now we're at the RC stage, which means we may still be able to take such changes, but they must have a significant value.

The .NET runtime provides a doc on what's considered a breaking change, what's allowed and what's not - https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md. According to the doc, adding new parameters is considered a breaking change and is not allowed, meaning we need to provide an overload:
![image](https://github.com/dotnet/extensions/assets/4403806/805f642e-faf6-43fa-8c82-c1bbbd2ca971)

I'm not sure how the defaultness affects this rule. @ericstj @bartonjs can you help?

Reply:
As I understand, both of the approaches are already not compatible with 1P services, since the methods are using IEnrichmentTagCollector instead of IEnrichmentPropertyBag, right? If so then I think we can stick to what is more convenient for users. IMO having one method is more convenient.

Reply:
You can still make breaking changes during RC.  @RussKie is correct that this type of change should not be done once the package is made stable.  Note that an additional method here will break interface implementers just as bad as changing the signature.  You should do your best to make sure your interface is complete in its first release.  Focus on what's best here to avoid future breaking changes and allow yourself some breaks now.

Reply:
> I think I'd rather stick to one method and add the optional exception.

In that case the `response` will be optional as well.

Reply:
Okay folks, I'm going to merge this as-is since Martin and Iliar both voted for having just one method.

Reply:
`request` is not optional, you're looking at the old version of the code. `response` and `exception` are both optional.

Reply:
![image](https://github.com/dotnet/extensions/assets/6381023/2b03c978-d2c8-458b-8b48-7367c42af975)

Reply:
@iliar-turdushev your comment was addressed in #4323 4323